pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -783,6 +787,49 @@ func (r *ReconcileClusterDeployment) updateClusterDeploymentStatus(cd *hivev1.Cl
 	return nil
 }
 
+func (r *ReconcileClusterDeployment) generateAdminKubeconfigSecret(cd *hivev1.ClusterDeployment, rawSecret *corev1.Secret, cdLog log.FieldLogger) (string, error) {
+	var hashPart string
+	if len(controllerutils.AdditionalCAHash()) == 0 {
+		hashPart = "original"
+	} else {
+		additionalCAHash := controllerutils.AdditionalCAHash()
+		hashPart = additionalCAHash[:20]
+	}
+	name := fmt.Sprintf("%s-%s", rawSecret.Name, hashPart)
+	secret := &corev1.Secret{}
+	err := r.Get(context.TODO(), types.NamespacedName{Namespace: rawSecret.Namespace, Name: name}, secret)
+	if err == nil {
+		return name, nil
+	}
+	if err != nil && !errors.IsNotFound(err) {
+		cdLog.WithError(err).WithField("name", name).Errorf("unexpected error fetching admin kubeconfig secret")
+		return "", err
+	}
+	updatedKubeconfig, err := controllerutils.FixupKubeconfig(rawSecret.Data["kubeconfig"])
+	if err != nil {
+		cdLog.WithError(err).Errorf("cannot fixup kubeconfig to generate new one")
+		return "", err
+	}
+	secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: rawSecret.Namespace,
+		},
+		Data: map[string][]byte{
+			"kubeconfig": updatedKubeconfig,
+		},
+	}
+	if err = controllerutil.SetControllerReference(cd, secret, r.scheme); err != nil {
+		cdLog.WithError(err).Error("error setting controller reference on kubeconfig secret")
+		return "", err
+	}
+	err = r.Create(context.TODO(), secret)
+	if err != nil {
+		cdLog.WithError(err).Errorf("cannot create admin kubeconfig secret")
+	}
+	return name, err
+}
```

Comment:
Now that this is in place, it looks like the other controllers are still using AdminKubeconfigSecret but still applying the FixupKubeconfig function to it. Do they need to be updated to either use this new secret without Fixup, or use Raw secret with fixup?

Reply:
Thanks, forgot about those. I'd prefer to update them to use the raw secret and fix them up, in case there is a delay in the clusterdeployment controller generating the updated secret.